packages/flame/lib/src/extensions/rect.dart

```diff
@@ -52,6 +53,19 @@ extension RectExtension on Rect {
     ];
   }
 
+  /// Transform Rect using the transformation defined by [matrix].
+  ///
+  /// **Note:** Rotation matrices will increase the size of the [Rect] but they
+  /// will not rotate it as [Rect] does not have any rotational values.
```

Comment:
I wonder if it would be best to validate and throw errors for invalid matrixes.

We could (in the future) also have a version that fully works with any matrix and returns a [Polygon](https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/geometry/polygon.dart) to represent non-AA rectangles

Reply:
I could, but not sure how to validate that on a matrix. So if anyone has any pointers, I am all ears!